inventory/group_vars/all.yml

```diff
@@ -94,14 +95,27 @@ postgres_shared_preload_libraries:
   - pg_stat_statements
   - wal2json
 
-postgresql_global_config_options:
+postgresql_global_config_defaults:
   - option: listen_addresses
     value: "{{ postgres_listen_addresses | join(',') }}"
   - option: shared_preload_libraries
     value: "{{ postgres_shared_preload_libraries | join(',') }}"
   - option: include_dir
     value: "conf.d"
 
+postgresql_global_config_custom: []
+
+postgresql_global_config_options: "{{ postgresql_global_config_defaults + postgresql_global_config_custom }}"
+
+postgres_host_options: "{{ '-h {{ db_host }}' if db_host != 'localhost' else '' }}"
+postgres_login_host: "{{ db_host if db_host != 'localhost' else omit }}"
+postgres_delegate_host: "{{ db_host if db_host != 'localhost' else inventory_hostname }}"
+
+dbserver_certs:
+  - postgresql.crt
+  - postgresql.key
+  - root.crt
```

Comment:
what are these? to accept SSL connections?

Reply:
Indeed! It's using self-signed certificates with a custom CA root key.

Reply:
but weren't using SSL already?

Reply:
Mostly we were not using SSL at all, just internal connections with peer authentication. The only case where we currently have connections from an external server is with Zapier, and we lock that down with a strong password and a single IP in the whitelist, for a single db user.

But that's using the generic certificates installed with postgres to handle the SSL, and they're probably not good enough for this use-case. Basically with encryption you have three parts: _Authentication_, _Authorisation_, and _Encryption in Transit_. The generic certs are only really good for the last one in that list. With custom certs we can do all three of those things for the connections between the app server and the db server, which will give vastly improved security.